packages/data/lib/src/repositories/auth_repository.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:domain/domain.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_facebook_auth/flutter_facebook_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+const _usersCollection = 'StarMovie_users';
+
+class AuthRepositoryImpl implements AuthRepository {
+  final FirebaseAuth firebaseAuth;
```

Comment:
why do we need FirebaseAuth?

Reply:
for saving user credential in to firebase auth. Should I remove it?

Reply:
no, keep it. seems that we need it